contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,245 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+
+import "../masterVault/interfaces/IMasterVault.sol";
+import "../bnbx/interfaces/IStakeManager.sol";
+import "./BaseStrategy.sol";
+
+contract BnbxYieldConverterStrategy is BaseStrategy {
+
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    IERC20Upgradeable private _bnbxToken;
+    IStakeManager private _stakeManager;
+    IMasterVault private _vault;
+
+    struct UserWithdrawRequest {
+        address recipient;
+        uint256 amount;
+        uint256 bnbxAmount;
+        uint256 triggerTime;
+    }
+    mapping(uint256 => UserWithdrawRequest) private _withdrawRequests;
+    uint256 private _firstWithdrawIdx;
+    uint256 private _nextWithdrawIdx;
+
+    uint256 private _bnbxHoldingBalance; // amount of bnbx held by this strategy
```

Comment:
Why do we need to keep track of this balance?
This is always `bnbXtoken.balanceOf(address.of(this)) - bnbxToUnstake`, isn't it?

We only track bnbxToUnstake and only bnbDeposited.

Reply:
yes the above should work unless
someone transfers bnbx token to this contract ?

Reply:
Done